pages/posts/dynamicref-and-generics.md

```diff
@@ -0,0 +1,172 @@
+
+However to support generics, we want to use this dynamic behavior a bit differently.  The strategy we'll use is a two-step process.
+
+1. For the generic type itself, we're going to write a schema with a reference that initially resolves to a subschema that always fails validation.
+2. For each derivation, we write a subschema that
+    - references the generic type schema from #1
+    - defines a new subschema for the same reference that describes the secondary type
+
+To see this in action, let's write schemas for the `List<T>` from above.  Then we'll write two more that will use it to help us define `List<string>` and `List<int>`.
+
+### The Generic Schema: `List<T>`
+
+We start simply with a list of things.
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://json-schema.example/list-of-t",
+  "type": "array"
+}
+```
+
+Now we define the items.  This is where `$dynamic*` does some work for us.
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://json-schema.blog/list-of-t",
+  "$defs": {
+    "content": {
+      "$dynamicAnchor": "T",
+      "not": true
+    }
+  },
+  "type": "array",
+  "items": { "$dynamicRef": "#T" }
+}
+```
+
+_**NOTE** I used `T` here to match up with the C# `List<T>` to better illustrate what's happening.  You can name it whatever you want._
+
+If we validate an instance against only this schema, `"$dynamicRef": "#T"` resolves to the subschema with `"$dynamicAnchor": "T"` that we included at `/$defs/content`.  In this case, `$dymamicRef` and `$dynamicAnchor` work just like `$ref` and `$anchor`.
+
+The `"not": true` means that all instances will fail validation.  Typically, to ensure all instances fail validation, we'd use the `false` schema, but in this case, we need to include a dynamic anchor, so a simple `false` doesn't work.  I think `"not": true` is probably the cleanest alternative, but you can also use something like `"allOf": [ false ]` if that makes more sense to you.
+
+_**NOTE** An empty array will still pass validation for this schema, but any array that has items will fail._
+
+You can also use multiple dynamic anchors to support types like `Dictionary<TKey, TValue>` that need multiple secondary types.
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://json-schema.blog/list-of-t",
+  "$defs": {
+    "key": {
+      "$dynamicAnchor": "TKey",
+      "not": true
+    },
+    "value": {
+      "$dynamicAnchor": "TValue",
+      "not": true
+    }
+  },
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties": {
+      "key": { "$dynamicRef": "#TKey" },
+      "key": { "$dynamicRef": "#TValue" }
```

Comment:
You have duplicate property names here. I think the second one was intended to be "value".